db/db_basic_test.cc

```diff
@@ -1204,19 +1204,32 @@ TEST_F(DBBasicTest, DBClose) {
   delete db;
   ASSERT_EQ(env->GetCloseCount(), 2);
 
+  // close by WaitForCompact() with close_db option
+  options.create_if_missing = false;
+  s = DB::Open(options, dbname, &db);
+  ASSERT_OK(s);
+  ASSERT_TRUE(db != nullptr);
+  WaitForCompactOptions wait_for_compact_options = WaitForCompactOptions();
+  wait_for_compact_options.close_db = true;
+  s = db->WaitForCompact(wait_for_compact_options);
+  ASSERT_EQ(env->GetCloseCount(), 3);
+  ASSERT_EQ(s, Status::IOError());
```

Comment:
TestLogger returns IOError upon Close() (Not sure why..)

https://github.com/facebook/rocksdb/blob/68a9cd21f2f6af4c6ab8724a19fbd4ea8ae89bdd/db/db_basic_test.cc#L1156-L1160

And this status replaces DB close status here
https://github.com/facebook/rocksdb/blob/68a9cd21f2f6af4c6ab8724a19fbd4ea8ae89bdd/db/db_impl/db_impl.cc#L711-L714

Reply:
Probably OK. Just add a comment like "see TestLogger::CloseHelper()"